### PR TITLE
feat: add  BaseSelect

### PR DESCRIPTION
**Pull request by tenadopt**

### CODE REVIEW

File: src/components/BaseSelect/BaseSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            const event = {
+                target: { value: option },
+            } as unknown as ChangeEvent<HTMLSelectElement>;
+
+            onChange(event);
+        }
+    };
+
+    const handleOpen = () => {
+        setOpen(!open);
```
Comment (shev7) [inline src/components/BaseSelect/BaseSelect.tsx]:
```suggestion
        setOpen((prev) => !prev);
```

File: src/components/BaseSelect/BaseSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
+import React, { ChangeEvent, useEffect, useRef, useState } from 'react';
+import { createPortal } from 'react-dom';
+import BaseInput from '../BaseInput/BaseInput';
+import BaseButton from '../BaseButton/BaseButton';
+import ArrowIcon from '../Icon/ArrowIcon';
```
Comment (shev7) [inline src/components/BaseSelect/BaseSelect.tsx]:
```suggestion
import ArrowIcon from 'assets/icons/arrow-icon.svg';
```

File: src/components/BaseSelect/BaseSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                    onClick={handleOpen}
+                    readOnly
+                >
+                    <button
+                        onClick={handleOpen}
+                        className={styles.selectButton}
+                        style={{ backgroundColor: 'transparent', borderColor: 'transparent' }}
+                    >
+                        <ArrowIcon className={arrow} />
+                    </button>
```
Comment (shev7) [inline src/components/BaseSelect/BaseSelect.tsx]: <IconButton/

File: src/components/BaseSelect/BaseSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+                </BaseInput>
+            </div>
+            {createPortal(
+                <div ref={divRef} className={optionsContainer}>
```
Comment (shev7) [inline src/components/BaseSelect/BaseSelect.tsx]: role listbox

File: src/utils/useOutsideClick.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { RefObject, useEffect } from 'react';
+
+export const useOutsideClick = (ref: RefObject<HTMLElement>, callback: () => void) => {
```
Comment (shev7) [inline src/utils/useOutsideClick.ts]:
```suggestion
export const useOutsideClick = <T extends HTMLElement>(callback: () => void) => {
```

**Review by shev7 — CHANGES_REQUESTED** [review]

File: src/components/BaseSelect/BaseSelect.tsx (hunk 1 above)
Comment (tenadopt) [reply src/components/BaseSelect/BaseSelect.tsx]: Done

File: src/components/BaseSelect/BaseSelect.tsx (hunk 2 above)
Comment (tenadopt) [reply src/components/BaseSelect/BaseSelect.tsx]: Done

File: src/components/BaseSelect/BaseSelect.tsx (hunk 4 above)
Comment (tenadopt) [reply src/components/BaseSelect/BaseSelect.tsx]: done

File: src/utils/useOutsideClick.ts (hunk 5 above)
Comment (tenadopt) [reply src/utils/useOutsideClick.ts]: Done

File: src/components/BaseButton/BaseButton.tsx [hunk 6]
```diff
@@ -12,18 +12,12 @@ export interface BaseButtonProps extends ComponentProps<'button'> {
...
     return (
-        <button
-            type="button"
-            className={btn}
-            disabled={props.disabled}
-            onClick={props.onClick}
-            {...props}
-        >
-            {props.children}
+        <button type="button" className={btn} {...props}>
```
Comment (shev7) [inline src/components/BaseButton/BaseButton.tsx]:
```suggestion
        <button type="button" className={btnClasses} {...props}>
```

File: src/components/Icon/IconButton.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import styles from '../BaseSelect/BaseSelect.module.scss';
+import { ReactComponent as Icon } from '../../assets/down-arrow.svg';
+
+const IconButton = ({ handleOpen, arrow }) => {
+    return (
+        <button
+            onClick={handleOpen}
+            className={styles.selectButton}
+            style={{ backgroundColor: 'transparent', borderColor: 'transparent' }}
```
Comment (shev7) [inline src/components/Icon/IconButton.tsx]: Avoid using inline styles

File: src/components/BaseButton/BaseButton.tsx (hunk 6 above)
Comment (tenadopt) [reply src/components/BaseButton/BaseButton.tsx]: Done

File: src/components/Icon/IconButton.tsx (hunk 7 above)
Comment (tenadopt) [reply src/components/Icon/IconButton.tsx]: Done

**Review by shev7 — APPROVED** [review]

**Pull request merged**